drivers/xbee/xbee.c

```diff
@@ -0,0 +1,700 @@
+    }
+    if (dev->event_cb != cb) {
+        return -ENOENT;
+    }
+    dev->event_cb = NULL;
+    return 0;
+}
+
+int _get(ng_netdev_t *netdev, ng_netconf_opt_t opt, void *value, size_t max_len)
+{
+    xbee_t *dev = (xbee_t *)netdev;
+    if (dev == NULL) {
+        return -ENODEV;
+    }
+
+    switch (opt) {
+        case NETCONF_OPT_ADDRESS:
+            return _get_addr_short(dev, (uint8_t *)value, max_len);
+        case NETCONF_OPT_ADDRESS_LONG:
+            return _get_addr_long(dev, (uint8_t *)value, max_len);
+        case NETCONF_OPT_CHANNEL:
+            return _get_channel(dev, (uint8_t *)value, max_len);
+        case NETCONF_OPT_NID:
+            return _get_panid(dev, (uint8_t *)value, max_len);
+        case NETCONF_OPT_PROTO:
+            return _get_proto(dev, (uint8_t *)value, max_len);
+        default:
+            return -ENOTSUP;
+    }
+}
+
+int _set(ng_netdev_t *netdev, ng_netconf_opt_t opt, void *value, size_t value_len)
+{
+    xbee_t *dev = (xbee_t *)netdev;
+    if (dev == NULL) {
+        return -ENODEV;
+    }
+
+    switch (opt) {
+        case NETCONF_OPT_ADDRESS:
+            return _set_addr(dev, (uint8_t *)value, value_len);
+        case NETCONF_OPT_CHANNEL:
+            return _set_channel(dev, (uint8_t *)value, value_len);
+        case NETCONF_OPT_NID:
+            return _set_panid(dev, (uint8_t *)value, value_len);
+        case NETCONF_OPT_PROTO:
+            return _set_proto(dev, (uint8_t *)value, value_len);
+        default:
+            return -ENOTSUP;
+    }
+}
+
+void _isr_event(ng_netdev_t *netdev, uint32_t event_type)
+{
+    xbee_t *dev = (xbee_t *)netdev;
+    ng_pktsnip_t *pkt_head;
+    ng_pktsnip_t *pkt;
+    ng_netif_hdr_t *hdr;
+    size_t pos;
+    size_t addr_len;
+    uint8_t cksum = 0;
+
+    /* check device */
+    if (dev == NULL) {
+        return;
+    }
+    /* check rx callback and event type */
+    if (event_type != ISR_EVENT_RX_DONE || dev->event_cb == NULL) {
+        dev->rx_count = 0;
+        return;
+    }
+
+    /* read address length */
+    if (dev->rx_buf[0] == API_ID_RX_SHORT_ADDR) {
+        addr_len = 2;
+    }
+    else {
+        addr_len = 8;
+    }
+
+    /* check checksum for correctness */
+    for (int i = 0; i < dev->rx_limit; i++) {
+        cksum += dev->rx_buf[i];
+    }
+    if (cksum != 0xff) {
+        DEBUG("xbee: Received packet with incorrect checksum, dropping it\n");
+        dev->rx_count = 0;
+        return;
+    }
+
+    /* allocate and fill interface header */
+    pkt_head = ng_pktbuf_add(NULL, NULL,
+                             sizeof(ng_netif_hdr_t) + (2 * addr_len),
+                             NG_NETTYPE_UNDEF);
+    if (pkt_head == NULL) {
+        DEBUG("xbee: Error allocating netif header in packet buffer on RX\n");
+        dev->rx_count = 0;
+        return;
+    }
+    hdr = (ng_netif_hdr_t *)pkt_head->data;
+    hdr->src_l2addr_len = (uint8_t)addr_len;
+    hdr->dst_l2addr_len = (uint8_t)addr_len;
```

Comment:
Don't know if this is correct. The address length in IEEE 802.15.4 can be of different lengths for source and destination